agent/iskakalunan.py

```diff
@@ -0,0 +1,350 @@
+# -*- coding: utf-8
+
+from __future__ import print_function
+import httplib2
+import urllib2
+import os
+import sys
+import re
+import json
+import base64
+import subprocess
+import shutil
+import zipfile
+import mimetypes
+import logging
+from email.mime.text import MIMEText
+from email.mime.base import MIMEBase
+from email.mime.multipart import MIMEMultipart
+
+from apiclient import discovery, errors
+from oauth2client import client
+from oauth2client import tools
+from oauth2client.file import Storage
+
+try:
+    import argparse
+    flags = argparse.ArgumentParser(parents=[tools.argparser]).parse_args()
+except ImportError:
+    flags = None
+
+SCOPES = 'https://www.googleapis.com/auth/gmail.modify'
+CLIENT_SECRET_FILE = 'client_secret.json'
+APPLICATION_NAME = 'Iskakalunan'
+
+LABEL_NEW_REQUEST = 'Label_8'
+LABEL_IN_QUEUE    = 'Label_4'
+LABEL_IN_PROCESS  = 'Label_5'
+LABEL_REPLIED     = 'Label_6'
+
+RESULT_ZIP = 'result.zip'
+RESULT_JSON = 'result.json'
+UPLOAD_FOLDER = 'upload'
+
+MSG_TYPE_INVALID_REQUEST = 'This is a notifier that your request is invalid.'
+MSG_TYPE_UNDER_EXECUTION = 'This is a notifier that your request is under execution.'
+MSG_TYPE_JOB_FINISHED    = 'This is a notifier that your request is finished.'
+
+logger = logging.getLogger('iskakalunan')
+logging.basicConfig(format='%(module)s %(levelname)s %(message)s', level=logging.DEBUG)
+
+
+class Iskakalunan(object):
+    def __init__(self):
+        """Basic Iskakalunan model
+
+        messages: mail queue on g-mail
+        """
+        self.credentials = self.get_credentials()
+        http = self.credentials.authorize(httplib2.Http())
+        self.service = discovery.build('gmail', 'v1', http=http)
+
+    def fetch_messages(self, label_id):
+        """Fetch messages from pre-set mail account
+
+        valid results will be updated to messages
+            - Valid result means messages contain 'Iskakalunan' (case insensitive)
+            - Only messages tags w/ 'NewRequest', 'InQueue', and 'InProcess'
+        """
+        try:
+            response = self.service.users().messages().list(userId='me',
+                                                       labelIds=label_id).execute()
+            message_list = []
+            messages = []
+            if 'messages' in response:
+                message_list.extend(response['messages'])
+
+            while 'nextPageToken' in response:
+                page_token = response['nextPageToken']
+                response = service.users().messages().list(userId=user_id,
+                                                           labelIds=label_id,
+                                                           pageToken=page_token).execute()
+                message_list.extend(response['messages'])
+
+            for message in message_list:
+                message_temp = Message(self.service, message['id'], message['threadId'])
+                if not 'moztpeqa' in message_temp.sender:
```

Comment:
same as another comment, should consider put this hard code into a global variable